opentelemetry-contrib/src/datadog/mod.rs

```diff
@@ -100,19 +140,24 @@ const DEFAULT_SERVICE_NAME: &str = "OpenTelemetry";
 /// Datadog span exporter
 #[derive(Debug)]
 pub struct DatadogExporter {
-    client: reqwest::Client,
-    request_url: Url,
+    client: Box<dyn HttpClient>,
+    request_url: Uri,
     service_name: String,
     version: ApiVersion,
 }
 
 impl DatadogExporter {
-    fn new(service_name: String, agent_endpoint: Url, version: ApiVersion) -> Self {
-        let mut request_url = agent_endpoint;
-        request_url.set_path(version.path());
+    fn new(
+        service_name: String,
+        agent_endpoint: Uri,
+        version: ApiVersion,
+        client: Box<dyn HttpClient>,
+    ) -> Self {
+        let request_url = agent_endpoint;
+        // request_url.set_path(version.path());
```

Comment:
Is this path not needed anymore?

Reply:
I think we already set path to version path [here](https://github.com/open-telemetry/opentelemetry-rust/pull/259/files#diff-3a90e7a53d74def8021c417e499b39296a43bca32ef38d94e0131676b2f8a804R213-R220). But yeah I should probably just use the param